### PR TITLE
Fix some metrics

### PR DESCRIPTION
**Pull request by huachaohuang**

### CODE REVIEW

**Comment by siddontang** [comment]
LGTM

File: src/util/rocksdb/engine_metrics.rs [hunk 1]
```diff
@@ -1009,11 +1009,18 @@ pub fn flush_engine_properties(engine: &DB, name: &str) {
...
+            .with_label_values(&[name])
+            .set(n as f64);
     }
     if let Some(t) = engine.get_property_int(ROCKSDB_OLDEST_SNAPSHOT_TIME) {
-        let d = time::get_time().sec.checked_sub(t as i64).unwrap_or(0);
-        STORE_ENGINE_OLDEST_SNAPSHOT_DURATION_GAUGE.set(d as f64);
+        // RocksDB returns 0 if no snapshots.
+        let now = time::get_time().sec;
+        if t > 0 && now > t {
+            STORE_ENGINE_OLDEST_SNAPSHOT_DURATION_GAUGE
```
Comment (BusyJay) [inline src/util/rocksdb/engine_metrics.rs]: Wrong name.

**Comment by huachaohuang** [comment]
PTAL @BusyJay

**Pull request merged**